nilearn/decomposition/base.py

```diff
@@ -0,0 +1,469 @@
+"""
+Base class for decomposition estimators, utilies for masking and reducing group
+data
+"""
+from __future__ import division
+
+from math import ceil
+
+import numpy as np
+from scipy import linalg
+from sklearn.base import BaseEstimator
+from sklearn.externals.joblib import Memory, Parallel, delayed
+from sklearn.linear_model import LinearRegression
+from sklearn.utils import check_random_state
+from sklearn.utils.extmath import randomized_svd
+
+from .._utils.cache_mixin import CacheMixin, cache
+from .._utils.niimg import _safe_get_data
+from .._utils.niimg_conversions import check_niimg_4d
+from ..input_data import NiftiMapsMasker
+from ..input_data.masker_validation import check_embedded_nifti_masker
+
+
+def mask_and_reduce(masker, imgs,
+                    confounds=None,
+                    reduction_ratio='auto',
+                    n_components=None, random_state=None,
+                    memory_level=0,
+                    memory=Memory(cachedir=None),
+                    n_jobs=1):
+    """Mask and reduce provided data with provided masker, using a PCA
```

Comment:
I don't think "using a PCA" is required because anyway you are describing below.